lib/ably/rest/channel.rb

```diff
@@ -103,7 +110,11 @@ def publish(name, data = nil, attributes = {})
           end
         end
 
-        response = client.post("#{base_path}/publish", payload.length == 1 ? payload.first : payload)
+        options = {}
+        if qs_params
+          options[:qs_params] = qs_params
```

Comment:
Oneliner like would be more readable, wdyt?
`options = { qs_params: qa_params } if qs_params`

Reply:
sure